tests/cypress/support/commands.js

```diff
@@ -38,16 +38,29 @@ Cypress.Commands.add(
     var user = OPTIONS_HUB_USER || Cypress.env('OPTIONS_HUB_USER')
     var password = OPTIONS_HUB_PASSWORD || Cypress.env('OPTIONS_HUB_PASSWORD')
     var idp = OPTIONS_HUB_OC_IDP || Cypress.env('OPTIONS_HUB_OC_IDP')
-    cy.visit('/multicloud/home/search')
-    cy.get('body').then((body) => {
-      // Check if logged in
-      if (body.find('#header').length === 0) {
-        // Check if identity providers are configured
-        if (body.find('form').length === 0) cy.contains(idp).click()
-        cy.get('#inputUsername').click().focused().type(user)
-        cy.get('#inputPassword').click().focused().type(password)
-        cy.get('button[type="submit"]').click()
-        cy.get('.pf-c-page__header', { timeout: 30000 })
+
+    cy.visit('/multicloud/home/welcome')
+
+    cy.url().then((res) => {
+      if (res.includes('oauth-openshift')) {
+        cy.log(
+          'The current user is logged out of the ACM console. Attempting to log into the console.'
+        )
+
+        cy.get('body').then((body) => {
+          // Check if logged in
+          if (body.find('#header').length === 0) {
+            // Check if identity providers are configured
+            if (body.find('form').length === 0) cy.contains(idp).click()
+
+            cy.get('#inputUsername').click().focused().type(user)
+            cy.get('#inputPassword').click().focused().type(password)
+            cy.get('button[type="submit"]').click()
+            cy.get('.pf-c-page__header')
+          }
+        })
+      } else {
+        cy.log('Confirmed that the user is logged. Procceding with the test.')
```

Comment:
The new `cy.login` flow will attempt to visit the welcome page. If the URL doesn't contain `oauth-openshift`, we know then that we are logged into the cluster console already; therefore, the login flow will be skipped and the test will proceed. If we are ever logged out unexpectedly, this will catch it and log us back in within the console.